File: ljh/level2/튜플/Tuple.java
/**
* 튜플
 */

import java.util.List;
import java.util.ArrayList;
import java.util.stream.*;

class Solution {
    public int[] solution(String s) {
        String[] intermediate = s.substring(0, s.length() - 1).split("}");
        List<Integer> answer = new ArrayList<>();
        
        Stream.of(intermediate)
            .map((str) -> str.substring(2))
            .map(str -> str.split(","))
            .map(arr -> Stream.of(arr).collect(Collectors.toList()))
            .sorted((a, b) -> a.size() - b.size())
            .forEach(list -> match(list, answer));
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
    
    public void match(List<String> list, List<Integer> answer) {
        for (String e : list) {
            Integer i = Integer.parseInt(e);
            if (answer.indexOf(i) == -1) {
                answer.add(i);
            }
        }
    } 
}

File: ljh/level2/튜플/Tuple.js
// 튜플
function solution(s) {
    const slice = [];
    const length = s.length - 1;
    let begin = 0;
    let end;
    while (begin !== length) {
        end = s.indexOf('}', begin);
        slice.push(s.substring(begin + 2 , end));
        begin = end + 1;
    }
    
    let arrSet = [];
    slice.forEach((value) => arrSet.push(value.split(',')));
    arrSet.sort((a, b) => a.length - b.length);
    
    let answer = [];
    for (let i = 0; i <arrSet.length; i++) {
        arrSet[i].forEach((value) => {
            value = parseInt(value);
            if(!answer.includes(value))
                answer.push(value);
        });
    }

    return answer;
}